src/Microsoft.AspNetCore.ResponseCaching/ResponseCache/IResponseCache.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.ResponseCaching
+{
+    public interface IResponseCache
+    {
+        IResponseCacheEntry Get(string key);
+        void Set(string key, IResponseCacheEntry entry);
```

Comment:
Would the implementation be simpler or harder if we had separate methods for the cache entry and the vary-by entry? Then you wouldn't need IResponseCacheEntry or downcasting.